### PR TITLE
Feature/animate bookmark

### LINKED ISSUE #62: Animated bookmarks
Use a cool "like" animation

### PR DESCRIPTION
**Pull request by emaccaull**
Closes #62

### CODE REVIEW

File: app/src/main/java/com/paranoidandroid/journey/recommendations/adapters/RecommendationsListAdapter.java [hunk 1]
```diff
@@ -152,11 +142,14 @@ private void configureFoursquareVenueViewHolder(RecommendationViewHolder vh, int
...
         if (this.listener != null) {
-            if (items.get(position).isBookmarked()) {
+            Recommendation rec = items.get(position);
+            if (rec.isBookmarked()) {
                 this.listener.onRemoveBookmarkClicked(items.get(position), position);
             } else {
                 this.listener.onAddBookmarkClicked(items.get(position), position);
             }
+            // Set immediately. Will be undone if save fails.
+            rec.setBookmarked(!rec.isBookmarked());
```
Comment (jkoryf) [inline app/src/main/java/com/paranoidandroid/journey/recommendations/adapters/RecommendationsListAdapter.java]: We are breaking encapsulation by directly modifying the recommendation object inside the adapter.

**Review by jkoryf — APPROVED** [review]
We are breaking encapsulation by directly modifying the recommendation object inside the adapter.

We can merge for now and make sure we fix this part.

**Comment by emaccaull** [comment]
Good point. I moved the logic to BaseRecommendationsFragment. Also reverted the ParseObject changes as they were causing an issue.

**Comment by jkoryf** [comment]
Looks good! :+1:

**Pull request merged**